i can only guess at what this test was trying to accomplish ...

ive submitted this upstream

--- main.c
+++ main.c
@@ -465,7 +465,7 @@
 	}
 	compute_segments(window);
 	window = 0;
-	for (v->pass_ticks=0, i=0; i<DEFTESTS != NULL; i++) {
+	for (v->pass_ticks=0, i=0; i<DEFTESTS; i++) {
 
 		/* Test to see if this test is selected for execution */
 		if (v->testsel >= 0) {
